diesel/src/connection/commit_error_processor.rs

```diff
@@ -0,0 +1,27 @@
+//! A module to evaluate what to do when a commit triggers an error.
+use crate::result::{DatabaseErrorKind, Error};
+
+/// Transaction status returned upon error on commit.
+#[derive(Debug)]
+pub enum CommitErrorOutcome {
```

Comment:
Maybe we should just use [`std::ops::ControlFlow`](https://doc.rust-lang.org/std/ops/enum.ControlFlow.html) here instead? On the other hand a custom enums makes it much clearer what happens for each variant.